addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -367,6 +359,15 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         const targetModelName = this.$target[0].dataset.model_name || 'mail.mail';
         this.activeForm = this.models.find(m => m.model === targetModelName);
         currentActionName = this.activeForm.website_form_label;
+
+        // Hide change form parameters option for forms
+        // e.g. User should not be enable to change existing job application form
+        // to opportunity form in 'Apply job' page.
+        this.modelCantChange = this.$target.attr('hide-change-model') !== undefined;
+        if (this.modelCantChange) {
+            return _super(...arguments);
+        }
```

Comment:
To avoid similar issues in the feature, what about nesting the `Create the Form Action select` lines within an `if` with the opposite condition, and avoid having two `return`s ?
This way if someone adds some other feature later, it will have less chances of accidentally being done only for non-`hide-change-model` forms.

Reply:
@bso-odoo Would you please explain me the benefit of doing this change, to be honest the issue is only tht super was getting called before `currentActionName` is defined, so maybe the fix looks good to me, still we are ready to change it if you have valid reason.

Reply:
@msh-odoo Well... the bug was introduced because https://github.com/odoo/odoo/pull/120874/commits/491a25b2baaa415a9bc26340eba5a11af8605411#diff-8b11d359e1213470895c7a9d28b45424520045089330be5713e9f3865ad16406R339 was done on the wrong side of the shortcut `return` which is moved in this PR. By writing the same fix in a way that keeps only a single `return`, such mistakes would automatically be avoided in the future. So I think this would be a more robust way to write the same code. Feel free to disagree, the current version if ok too... I just think we can use the opportunity to make it more robust. Or maybe we can do that change in master only - since such a mistake would typically only happen in an `[IMP]`.

Reply:
@bso-odoo My bad, I misunderstood your comment, we have improved it again.